rotkehlchen/api/rest.py

```diff
@@ -1206,28 +1207,53 @@ def user_login(
         # up until here. Even with non valid keys in the DB login should work fine
         except AuthenticationError as e:
             self.rotkehlchen.reset_after_failed_account_creation_or_login()
-            result_dict['message'] = str(e)
-            return api_response(result_dict, status_code=HTTPStatus.UNAUTHORIZED)
+            return wrap_in_fail_result(
+                message=str(e),
+                status_code=HTTPStatus.UNAUTHORIZED,
+            )
         except RotkehlchenPermissionError as e:
             self.rotkehlchen.reset_after_failed_account_creation_or_login()
-            result_dict['result'] = e.message_payload
-            result_dict['message'] = e.error_message
-            return api_response(result_dict, status_code=HTTPStatus.MULTIPLE_CHOICES)
+            return {
+                'result': e.message_payload,
+                'message': e.error_message,
+                'status_code': HTTPStatus.MULTIPLE_CHOICES,
+            }
         except (DBUpgradeError, SystemPermissionError, DBSchemaError) as e:
             self.rotkehlchen.reset_after_failed_account_creation_or_login()
-            result_dict['message'] = str(e)
-            return api_response(result_dict, status_code=HTTPStatus.CONFLICT)
+            return wrap_in_fail_result(
+                message=str(e),
+                status_code=HTTPStatus.CONFLICT,
+            )
         except sqlcipher.OperationalError as e:  # pylint: disable=no-member
             self.rotkehlchen.reset_after_failed_account_creation_or_login()
-            result_dict['message'] = f'Unexpected database error: {str(e)}'
-            return api_response(result_dict, status_code=HTTP_STATUS_INTERNAL_DB_ERROR)  # type: ignore  # noqa: E501
+            return wrap_in_fail_result(
+                message=f'Unexpected database error: {str(e)}',
+                status_code=HTTP_STATUS_INTERNAL_DB_ERROR,  # type: ignore
+            )
+
         # Success!
+        exchanges = self.rotkehlchen.exchange_manager.get_connected_exchanges_info()
         with self.rotkehlchen.data.db.conn.read_ctx() as cursor:
-            result_dict['result'] = {
-                'exchanges': self.rotkehlchen.exchange_manager.get_connected_exchanges_info(),
-                'settings': process_result(self.rotkehlchen.get_settings(cursor)),
-            }
-        return api_response(result_dict, status_code=HTTPStatus.OK)
+            settings = process_result(self.rotkehlchen.get_settings(cursor))
+
+        return _wrap_in_ok_result({
+            'exchanges': exchanges,
+            'settings': settings,
+        })
+
+    def user_login(
+            self,
+            name: str,
+            password: str,
+            sync_approval: Literal['yes', 'no', 'unknown'],
+    ) -> Response:
+        # Login is always async
```

Comment:
Why make it always async though? Can't you give the option, like we do for all other endpoints?

I would **strongly** prefer that. Does not matter in production, but in testing, having the option to log in with just a single api call is good. Undoing that will make all our manual testing work more complicated.

```suggestion
       """The user login process is always async"""
```